File: ABC121/b-train.cpp
#include <iostream>
#include <vector>

int solve(int n, int m, int c, const std::vector<int> & B, const std::vector<std::vector<int>*> & A) {
    int ans = 0;
    for (int i = 0; i < n; i++) {
        int sum = c;
        for (int j = 0; j < m; j++) {
            sum += A[i]->at(j) * B[j];
        }
        ans += sum > 0;
    }
    return ans;
}

int main() {
    int n, m, c;
    std::cin >> n >> m >> c;
    std::vector<int> B(m, 0);
    for (int i = 0; i < m; i++) {
        std::cin >> B[i];
    }
    std::vector<std::vector<int>*> A(n, nullptr);
    for (int i = 0; i < n; i++) {
        A[i] = new std::vector<int>(m, 0);
        for (int j = 0; j < m; j++) {
            std::cin >> A[i]->at(j);
        }
    }
    std::cout << solve(n, m, c, B, A) << std::endl;
}

File: ABC094/b-train.rs
use std::io;

fn read_line(cin: &io::Stdin) -> String {
    let mut s = String::new();
    cin.read_line(&mut s).unwrap();
    s.trim().to_string()
}

fn main() {
    let cin = io::stdin();
    let (n, x) = {
        let v: Vec<usize> = read_line(&cin)
            .split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();
        (v[0], v[2])
    };
    let a: Vec<usize> = read_line(&cin)
        .split_whitespace().map(|x| x.parse().unwrap()).collect();
    let mut costs = vec![0; n];
    for &i in a.iter() {
        costs[i] += 1;
    }
    let cost_l = (0 .. x)
        .map(|i| costs[i])
        .sum::<i32>();
    let cost_r = (x + 1 .. n)
        .map(|i| costs[i])
        .sum::<i32>();
    let ans = std::cmp::min(cost_l, cost_r);
    println!("{}", ans);
}